browser/ui/views/brave_actions/brave_shields_action_view.cc

```diff
@@ -215,11 +225,20 @@ void BraveShieldsActionView::OnTabStripModelChanged(
     TabStripModel* tab_strip_model,
     const TabStripModelChange& change,
     const TabStripSelectionChange& selection) {
+  if (change.type() == TabStripModelChange::kRemoved) {
+    auto* web_contents = tab_strip_model->GetActiveWebContents();
+    if (web_contents) {
+      brave_shields::BraveShieldsDataController::FromWebContents(web_contents)
+          ->RemoveObserver(this);
+    }
+  }
+
   if (selection.active_tab_changed()) {
     if (selection.new_contents) {
       brave_shields::BraveShieldsDataController::FromWebContents(
           selection.new_contents)
           ->AddObserver(this);
+      active_web_contents_ = selection.new_contents->GetWeakPtr();
```

Comment:
if new_contents is null we should set `active_web_contents_ = base::WeakPtr<WebContents>()`